modules/ai/assets/js/editor/pages/form-text/index.js

```diff
@@ -52,14 +53,30 @@ const FormText = (
 		additionalOptions,
 		credits,
 		usagePercentage,
+		promptHistoryAction,
 	},
 ) => {
 	const initialValue = getControlValue() === additionalOptions?.defaultValue ? '' : getControlValue();
 
-	const { data, isLoading, error, reset, send, sendUsageData } = useTextPrompt( { result: initialValue, credits } );
+	const { data, isLoading, error, setResult, reset, send, sendUsageData } = useTextPrompt( { result: initialValue, credits } );
 
 	const [ prompt, setPrompt ] = useState( '' );
 
+	useEffect( () => {
+		if ( ! promptHistoryAction.type ) {
+			return;
+		}
+
+		if ( promptHistoryAction.type === HISTORY_ACTION_TYPES.REUSE ) {
+			reset();
+			setPrompt( promptHistoryAction.data );
+		}
+
+		if ( promptHistoryAction.type === HISTORY_ACTION_TYPES.EDIT ) {
+			setResult( promptHistoryAction.data );
+		}
+	}, [ promptHistoryAction ] );
```

Comment:
Tbh, I don't see anything weird here. 

The Form components are "subscribing" to a particular change (either reuse, restore or edit).

The context was created in the `<History />` component just to prevent moving data/methods to the deep nested components via props. So `<PageContent />` or `<PromptDialog />` knows nothing about it.

Moreover, creating a context just to pass state to the next level of children might add some level of abstraction but in its essence will be the same thing. However, I will consider moving it to a higher level as I now can see it might allow me to remove the `isOpen` state.

P. S. A Friendly recommendation from me would be to read [this document](https://chromium.googlesource.com/chromium/src/+/master/docs/cr_respect.md#don_t-use-extreme-or-very-negative-language) to be even more effective reviewer.

Particularly, I'd avoid sentences like this:

> This concept is weird..

> try to think of a better solution.

They bring no added value but might be a reason for frustration for some.